### DISCUSSION TITLE
Columns modifications using "column_reflect" event for Base.metadata using automap doesn't reflect changes in Base.classes

### DISCUSSION THREAD

**Discussion by gunzino** [sqlalchemy/sqlalchemy#10521] [Usage Questions] [answered]
### Describe the bug

Hello,

I use automap functionality to generate the proper classes from tables. I would like to add generic-like solution for 'modified_timestamp' columns in tables. I'm trying to use "column_reflect" event and modify the "onupdate" value. However, the changes are not then reflected in Base.classes, probably only in Base.metadata. Is it the intended behavior ? Or is there some other approach how to achieve this?

Thanks!

### SQLAlchemy Version in Use

2.0

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

PostgreSQL 15

### Python Version

3.10

### Operating system

Linux

### To Reproduce

```python
@listens_for(Base.metadata, "column_reflect")
def column_reflect(inspector, table, column_info):
  if column_info['name'] == "modified_timestamp":
    column_info['onupdate'] = func.current_timestamp()
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
```

**Comment by CaselIT** [comment]
Hi,

That's expected, since only the known keys of `column_info` are used. onupdate it's not one of the default keys so that's not copied over on the actual column that's created.

I'm not sure if there is an event that can be used for this. @zzzeek do you have suggestions?

**Reply by zzzeek** [reply]
it's possible to use the [before_parent_attach](https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.DDLEvents.before_parent_attach) .  you'd apply this to `Column` and it would be called with the `Column` and `Table` every time `Column` is associated with a `Table`, but you'd need to use private API to do it correctly, as `ColumnDefault(<arg>, onupdate=True)._set_parent(column)`.

the `column_reflect` allows one to set the `default` for the column which can be given a Python value or callable, there does not seem to be a restriction there.  Therefore I'd consider this a bug in fact that we aren't allowing "onupdate" in the dictionary, there's no reason why we shouldn't support that here.

**Reply by gunzino** [reply, marked as answer]
@zzzeek 
Thanks for leading me to **before_parent_attach** and **ColumnDefault.** I think you meant to use **for_update=True** instead of **onupdate=True** like ``ColumnDefault(<arg>, for_update=True)._set_parent(column)``.

The working approach seems to be:
```
@listens_for(Column, "before_parent_attach")
def receive_before_parent_attach(target, parent):
    if (target.name == "modified_timestamp"):
        target.onupdate = ColumnDefault(func.current_timestamp())
```

Another approach - no difference in functionality - I would be grateful if you explain if there is some difference
```
@listens_for(Column, "before_parent_attach")
def receive_before_parent_attach(target, parent):
    if (target.name == "modified_timestamp"):
        ColumnDefault(func.current_timestamp(), for_update=True)._set_parent(target)
```

**Comment by CaselIT** [comment]
> the column_reflect allows one to set the default for the column which can be given a Python value or callable, there does not seem to be a restriction there. Therefore I'd consider this a bug in fact that we aren't allowing "onupdate" in the dictionary, there's no reason why we shouldn't support that here.

is the proposed change to take any arbitrary key from the dict and apply them to the column that's created?
